bot/zephyr.py

```diff
@@ -87,26 +87,29 @@ def source_zephyr_env(zephyr_wd):
     return env
 
 
-def build_and_flash(zephyr_wd, board, conf_file=None):
+def build_and_flash(zephyr_wd, board, board_id, conf_file=None):
     """Build and flash Zephyr binary
     :param zephyr_wd: Zephyr source path
     :param board: IUT
+    :param board_id: Segger ID
     :param conf_file: configuration file to be used
     :return: TTY path
     """
-    logging.debug("{}: {} {} {}". format(build_and_flash.__name__, zephyr_wd,
-                                         board, conf_file))
+    logging.debug("{}: {} {} {} {}". format(build_and_flash.__name__,
+                                            zephyr_wd, board,
+                                            board_id, conf_file))
     tester_dir = os.path.join(zephyr_wd, "tests", "bluetooth", "tester")
 
     # Set Zephyr project env variables
     env = source_zephyr_env(zephyr_wd)
 
-    cmd = ['west',  'build', '-p', 'auto', '-b', board]
+    cmd_build = ['west',  'build', '-p', 'auto', '-b', board]
     if conf_file:
-        cmd.extend(('--', '-DCONF_FILE={}'.format(conf_file)))
+        cmd_build.extend(('--', '-DCONF_FILE={}'.format(conf_file)))
+    check_call(cmd_build, env=env, cwd=tester_dir)
 
-    check_call(cmd, env=env, cwd=tester_dir)
-    check_call(['west', 'flash'], env=env, cwd=tester_dir)
+    cmd_flash = ['west', 'flash', '--snr', board_id]
```

Comment:
What if the `board_id` is empty?

Reply:
before changes (when it was only west flash) and no device attached. It should return the same error.

This case when there is more than one device connected, it should find the correct one to flash.

Do you want to add control if board_id is empty?

Reply:
Yes, I want to keep old behavior if `board_id` is empty.

Reply:
updated the PR also covering the empty cases